dagger/tasks/task.py

```diff
@@ -945,7 +945,11 @@ class INonLeafNodeTask(ITask[KT, VT], abc.ABC):
     task_type: str = TaskType.SUB_DAG.name
 
     async def stop(self) -> None:
-        pass
+        self.status = TaskStatus(
+            code=TaskStatusEnum.STOPPED.name,
+            value=TaskStatusEnum.STOPPED.value,
+        )
+        await super().stop()
```

Comment:
Is this necessary if the parent will set the status of the child tasks?

Reply:
DWR: You could move the logic for checking status here and just call task.stop() on all remaining tasks in the parent dag - not sure why but that seems cleaner to me 😅 
```
    async def stop(self) -> None:
        if task.status.code not in TERMINAL_STATUSES:
          self.status = TaskStatus(
              code=TaskStatusEnum.STOPPED.name,
              value=TaskStatusEnum.STOPPED.value,
          )
        if task.status.code == TaskStatusEnum.EXECUTING.name:
           await super().stop()
```
And then just 
```
        if remaining_tasks:
            for task in remaining_tasks:
                task.stop()
```